src/main/java/org/sonarsource/sonarlint/ls/SonarLintLanguageServer.java

```diff
@@ -803,4 +807,55 @@ public CompletableFuture<SonarLintExtendedLanguageClient.ShowRuleDescriptionPara
     var ruleContextKey = Objects.isNull(issue) ? "" : issue.getIssue().getRuleDescriptionContextKey().orElse("");
     return commandManager.getShowRuleDescriptionParams(fileUri, ruleKey, ruleContextKey);
   }
+
+  @Override
+  public CompletableFuture<Void> changeHotspotStatus(ChangeHotspotStatusParams params) {
+    var workspace = workspaceFoldersManager.findFolderForFile(create(params.getFileUri()))
+      .orElseThrow(() -> new IllegalStateException("No workspace found"));
+    var workspaceUri = workspace.getUri();
+    var hotspotStatus = hotspotStatusOfTitle(params.getNewStatus());
+    var coreParams = new org.sonarsource.sonarlint.core.clientapi.backend.hotspot.ChangeHotspotStatusParams(
+      workspaceUri.toString(), params.getHotspotKey(), hotspotStatus);
+    return backendServiceFacade.getBackendService().changeHotspotStatus(coreParams).thenAccept(nothing -> {
+      var key = params.getHotspotKey();
+      if (hotspotStatus != HotspotStatus.TO_REVIEW && hotspotStatus != HotspotStatus.ACKNOWLEDGED) {
+        securityHotspotsCache.removeIssueWithServerKey(params.getFileUri(), key);
+      } else {
+        securityHotspotsCache.updateIssueStatus(params.getFileUri(), key, hotspotStatus);
+      }
+      diagnosticPublisher.publishDiagnostics(create(params.getFileUri()), true);
+      client.showMessage(new MessageParams(MessageType.Info, "Hotspot status was successfully changed"));
+    }).exceptionally(t -> {
+      lsLogOutput.error("Error changing hotspot status", t);
+      client.showMessage(new MessageParams(MessageType.Error, "Could not change status for the hotspot. Look at the SonarLint output for details."));
+      return null;
+    });
+  }
+
+  @Override
+  public CompletableFuture<GetAllowedHotspotStatusesResponse> getAllowedHotspotStatuses(GetAllowedHotspotStatusesParams params) {
+    var folderUri = params.getFolderUri();
+    var bindingOptional = bindingManager.getBinding(create(folderUri));
+    if (bindingOptional.isEmpty()) {
+      return CompletableFuture.completedFuture(null);
+    }
+    var connectionId = bindingOptional.get().getConnectionId();
+    var checkStatusChangePermittedParams = new CheckStatusChangePermittedParams(connectionId, params.getHotspotKey());
+    return backendServiceFacade.getBackendService().getAllowedHotspotStatuses(checkStatusChangePermittedParams).thenApply(r -> {
+        var delegatingIssue = (DelegatingIssue) securityHotspotsCache
+          .findIssueWithId(params.getFileUri(), params.getHotspotKey()).get().getValue().getIssue();
+        var reviewStatus = delegatingIssue.getReviewStatus();
+        var statuses = r.getAllowedStatuses().stream().filter(s -> s != hotspotStatusValueOfHotspotReviewStatus(reviewStatus))
```

Comment:
Why is this util method needed? i.e. why are there two enums with the same values in core? It is confusing, but I understand not directly related to this PR.